cfme/tests/openstack/infrastructure/test_host_power_control.py

```diff
@@ -17,9 +17,9 @@ def host_collection(appliance):
 
 
 @pytest.fixture(scope='module')
-def host_on(host_collection, provider):
+def host_on(host_collection):
     try:
-        my_host_on = provider.nodes.all().pop()
```

Comment:
What is the purpose of this change? Now the host that we return _may not_ be associated with the provider for the test function.

Reply:
@dkholodo: I like @mshriver comment. Suppose, we have already have one provider lets `vsphere` then we add `rhos uc` then about situation will occur means `host_collection.all()` will return you all host for both provider.

I would like to suggest you; Please, use `filtering` of `all` method. It will help you to filter out `hosts/nodes` for specific provider and then you can select anyone of them for power operation.